### PR TITLE
fix(react): form onSubmit values

### PR DESCRIPTION
**Pull request by rabelloo**
## Purpose

The current approach will read `id` off elements if they have one, instead of reading their `names`.

That's because of how `<form>.elements` work.

image

## Approach

Instead of using the `elements` key, just list them as an array and filter the ones that have the `name` property set.
It's actually much more efficient this way, and we can remove some code.

## Testing

Updated unit tests.

## Risks

None.

### CODE REVIEW

File: packages/react/src/components/form/getFormValues.ts [hunk 1]
```diff
@@ -13,14 +13,10 @@
  */
 export const getFormValues = <T extends Result>(form: HTMLFormElement): T =>
   Object.fromEntries(
-    Object.entries(form.elements).filter(hasName).map(elementValue)
-  );
+    (Array.from(form.elements) as HTMLInputElement[])
```
Comment (josokinas) [inline packages/react/src/components/form/getFormValues.ts:16]: If you set this as `HTMLInputElement`, does that work with other form elements?

**Review by josokinas — APPROVED** [review]

File: packages/react/src/components/form/getFormValues.ts (hunk 1 above)
Comment (rabelloo) [reply packages/react/src/components/form/getFormValues.ts:16]: It does, all we need is elements with `.value`, all all `form.elements` should have them.
It's not the most precise or correct interface/type to use, but it's not terribly wrong either.
I suppose we could do `as { value: string }` 🤷🏻

Comment (josokinas) [reply packages/react/src/components/form/getFormValues.ts:16]: Maybe it woul be better?

IDK, just seems that you only expect `<input>` here...

Comment (rabelloo) [reply packages/react/src/components/form/getFormValues.ts]: Fair enough, created a new interface.

**Pull request merged**